test/python/algorithms/test_estimator_gradient.py

```diff
@@ -0,0 +1,330 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2019, 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+# =============================================================================
+
+""" Test Quantum Gradient Framework """
+
+import unittest
+from test import combine
+
+import numpy as np
+from ddt import ddt
+
+from qiskit import QuantumCircuit
+from qiskit.algorithms.gradients import (
+    FiniteDiffEstimatorGradient,
+    ParamShiftEstimatorGradient,
+    LinCombEstimatorGradient,
+    SPSAEstimatorGradient,
+)
+from qiskit.circuit import Parameter
+from qiskit.circuit.library import EfficientSU2, RealAmplitudes
+from qiskit.exceptions import QiskitError
+from qiskit.primitives import Estimator
+from qiskit.quantum_info import SparsePauliOp
+from qiskit.test import QiskitTestCase
+
+
+@ddt
+class TestEstimatorGradient(QiskitTestCase):
```

Comment:
Could you add some tests for the SPSA gradients too? 🙂

Reply:
Actually, tests for the SPSA gradients are at the last of `TestEstimatorGradient`. I could not put it in `    @combine(
        grad=[FiniteDiffEstimatorGradient, ParamShiftEstimatorGradient, LinCombEstimatorGradient]
    )` because the gradient value is too different. 
https://github.com/a-matsuo/qiskit-terra/blob/9abbd2ac79ef1177308c236c0be344d2333384c6/test/python/algorithms/test_estimator_gradient.py#L302